src/Tasks/Hash.cs

```diff
@@ -18,7 +18,19 @@ namespace Microsoft.Build.Tasks
     /// </remarks>
     public class Hash : TaskExtension
     {
-        private const char ItemSeparatorCharacter = '\u2028';
+        private const char s_itemSeparatorCharacter = '\u2028';
+
+        private static readonly Encoding s_encoding = Encoding.UTF8;
+
+        private static readonly byte[] s_itemSeparatorCharacterBytes = s_encoding.GetBytes(new char[] { s_itemSeparatorCharacter });
+
+        // Size of buffer where bytes of the strings are stored until sha1.TransformBlock is be run on them.
+        // It is needed to get a balance between amount of costly sha1.TransformBlock calls and amount of allocated memory.
+        private const int sha1BufferSize = 512;
+
+        // Size of chunks in which ItemSpecs would be cut.
+        // String of size 169 gives no more than ~512 bytes in utf8 encoding.
```

Comment:
NIT: Please consider to include further clarification, something like: `We have chosen this length so byteBuffer rented from ArrayPool will be close but not bigger than 512."